File: frontend/src/main/js/composables/model/ResultMessage.js
export class ResultMessage {
  constructor() {
    this._type = undefined;
    this._message = undefined;
    this._details = undefined;
  }

  get type() {
    return this._type;
  }

  get message() {
    return this._message;
  }

  get details() {
    return this._details;
  }

  get color() {
    if (this._type === 'ERROR' || this._type === 'WARNING') {
      return 'is-danger';
    } else if (this._type === 'INFO') {
      return 'is-success';
    } else {
      return 'is-info';
    }
  }

  get hasMessage() {
    return this._message !== undefined;
  }

  get hasDetails() {
    if (!this.hasMessage) return false;

    return this._details !== undefined;
  }

  set(type = '', message = '', details = '') {
    this._type = type !== '' ? type : undefined;
    this._message = message !== '' ? message : undefined;
    this._details = details !== '' ? details : undefined;
  }
}

File: src/main/kotlin/example/domain/shared/message/ResultMessage.kt
package example.domain.shared.message

/**
 * 処理実行結果をユーザに通知するために使用する
 */
data class ResultMessage private constructor(val type: Type,
                                             val message: String,
                                             val details: String?) {
    enum class Type {
        /**
         * ユーザの操作による処理が正常に実行された場合の結果メッセージのタイプ
         */
        INFO,

        /**
         * 何らかの問題があるが、処理を続行できる場合の結果メッセージのタイプ
         */
        WARNING,

        /**
         * 業務ロジックでのエラーやシステム起因のエラーと判定された場合の結果メッセージのタイプ
         */
        ERROR
    }

    companion object {
        fun info(message: String, details: String? = null): ResultMessage {
            return ResultMessage(Type.INFO, message, details)
        }

        fun warning(message: String, details: String? = null): ResultMessage {
            return ResultMessage(Type.WARNING, message, details)
        }

        fun error(message: String, details: String? = null): ResultMessage {
            return ResultMessage(Type.ERROR, message, details)
        }
    }
}